src/hooks/project-hooks.tsx

```diff
@@ -182,17 +229,25 @@ export const ProjectProvider = ({
       getCurrentProject,
       expectChangedHeader,
       projectFlushedToEditor,
+      checkIfEditorStartUpTimedOut,
+      editorTimedOut,
     ]
   );
   const openEditor = useCallback(async () => {
     logging.event({
       type: "edit-in-makecode",
     });
-    await doAfterEditorUpdate(() => {
-      navigate(createCodePageUrl());
-      return Promise.resolve();
-    });
-  }, [doAfterEditorUpdate, logging, navigate]);
+    try {
+      await doAfterEditorUpdate(() => {
+        navigate(createCodePageUrl());
+        return Promise.resolve();
+      });
+    } catch (e) {
+      if (e instanceof CodeEditorError) {
+        openEditorTimedOutDialog();
+      }
+    }
```

Comment:
Found it better to handle with try-catch seeing as we don't want a `openEditorTimedOutDialog` dialog to pop up for `browserNavigationToEditor` (see function below)